easybuild/easyblocks/l/lammps.py

```diff
@@ -0,0 +1,362 @@
+        # see: https://github.com/lammps/lammps/issues/1110
+        if '-DEIGEN3_INCLUDE_DIR=' not in self.cfg['configopts']:
+            if get_software_root('Eigen'):
+                self.cfg.update('configopts', '-DEIGEN3_INCLUDE_DIR=%s/include/Eigen' % get_software_root('Eigen'))
+
+        if '-DEigen3_DIR=' not in self.cfg['configopts']:
+            if get_software_root('Eigen'):
+                self.cfg.update('configopts', '-DEigen3_DIR=%s/share/eigen3/cmake/' % get_software_root('Eigen'))
+
+        # LAMMPS Configuration Options
+        # https://github.com/lammps/lammps/blob/master/cmake/README.md#lammps-configuration-options
+        if self.cfg['general_packages']:
+            for package in self.cfg['general_packages']:
+                self.cfg.update('configopts', '-DPKG_%s=on' % package)
+
+        if self.cfg['user_packages']:
+            for package in self.cfg['user_packages']:
+                self.cfg.update('configopts', '-DPKG_USER-%s=on' % package)
+
+        # Optimization settings
+        if '-DPKG_OPT=' not in self.cfg['configopts']:
+            self.cfg.update('configopts', '-DPKG_OPT=on')
+
+        if '-DPKG_USR-INTEL=' not in self.cfg['configopts']:
+            self.cfg.update('configopts', '-DPKG_USER-INTEL=on')
+
+        # MPI/OpenMP
+        if self.toolchain.options.get('usempi', None):
+            self.cfg.update('configopts', '-DBUILD_MPI=yes')
+        if self.toolchain.options.get('openmp', None):
+            self.cfg.update('configopts', '-DBUILD_OMP=yes')
+            self.cfg.update('configopts', '-DPKG_USER-OMP=on')
+
+        # FFT
+        if '-DFFT=' not in self.cfg['configopts']:
+            self.cfg.update('configopts', '-DFFT=FFTW3')
+        if '-DFFT_PACK=' not in self.cfg['configopts']:
+            self.cfg.update('configopts', '-DFFT_PACK=array')
+
+        # https://lammps.sandia.gov/doc/Build_extras.html
+        # KOKKOS
+        if self.cfg['kokkos']:
+
+            if self.toolchain.options.get('openmp', None):
+                self.cfg.update('configopts', '-DKOKKOS_ENABLE_OPENMP=yes')
+
+            self.cfg.update('configopts', '-DPKG_KOKKOS=on')
+            self.cfg.update('configopts', '-DKOKKOS_ARCH="%s"' % self.get_kokkos_gpu_arch(cuda_cc))
+
+            # if KOKKOS and CUDA
+            if cuda:
+                self.check_cuda_compute_capabilities(cfg_cuda_cc, ec_cuda_cc, cuda_cc)
+                nvcc_wrapper_path = os.path.join(self.start_dir, "lib", "kokkos", "bin", "nvcc_wrapper")
+                # nvcc_wrapper can't find OpenMP for some reason, need to specify OpenMP flags.
+                # https://software.intel.com/en-us/forums/intel-oneapi-base-toolkit/topic/842470
+                self.cfg.update('configopts', '-DOpenMP_CXX_FLAGS="-qopenmp"')
+                self.cfg.update('configopts', '-DOpenMP_CXX_LIB_NAMES="libiomp5"')
+                self.cfg.update('configopts', '-DOpenMP_libiomp5_LIBRARY=$EBROOTICC/lib/intel64_lin/libiomp5.so')
```

Comment:
This is clearly toolchain specific behaviour so you will check the toolchain so that the behaviour is valid.

Reply:
@boegel Do we have generic variables we can leverage for this case?

Reply:
I think we do, LIB_MULTITHREAD

Reply:
Good catch :) Just found out that this might not be needed - might be an issue with intel compilers not finding stuff. Trying workaround.

Reply:
unsetting LIBS in easyconfig resolves the issue with compiler having trouble finding stuff here.